ts/features/bonusVacanze/store/sagas/activation/__tests__/integration/activationIntegrationTests.ts

```diff
@@ -0,0 +1,134 @@
+import { right } from "fp-ts/lib/Either";
+import { fromNullable } from "fp-ts/lib/Option";
+import { Action, combineReducers } from "redux";
+import { expectSaga } from "redux-saga-test-plan";
+import * as matchers from "redux-saga-test-plan/matchers";
+import { select } from "redux-saga-test-plan/matchers";
+import { navigateToWalletHome } from "../../../../../../../store/actions/navigation";
+import { navigationHistoryPop } from "../../../../../../../store/actions/navigationHistory";
+import { navigationCurrentRouteSelector } from "../../../../../../../store/reducers/navigation";
+import BONUSVACANZE_ROUTES from "../../../../../navigation/routes";
+import {
+  cancelBonusRequest,
+  completeBonusVacanzeActivation
+} from "../../../../actions/bonusVacanze";
+import allActiveReducer from "../../../../reducers/allActive";
+import { bonusActivationSaga } from "../../getBonusActivationSaga";
+import { handleBonusActivationSaga } from "../../handleBonusActivationSaga";
+import bonusVacanzeActivationReducer, {
+  BonusActivationProgressEnum
+} from "./../../../../reducers/activation";
+import {
+  ActivationBackendResponse,
+  backendIntegrationTestCases,
+  MockActivationState
+} from "./mockData";
+
+jest.mock("react-native-background-timer", () => {
+  return {
+    startTimer: jest.fn()
+  };
+});
+
+jest.mock("react-native-share", () => {
+  return {
+    open: jest.fn()
+  };
+});
+
+const activationReducer = combineReducers<MockActivationState, Action>({
+  activation: bonusVacanzeActivationReducer,
+  allActive: allActiveReducer
+});
+
+describe("Bonus Activation Saga Integration Test", () => {
+  it("Cancel A bonus request after server error", () => {
+    const startBonusActivation = jest.fn();
+    const getActivationById = jest.fn();
+
+    return expectSaga(
+      handleBonusActivationSaga,
+      bonusActivationSaga(startBonusActivation, getActivationById)
+    )
+      .withReducer(activationReducer)
+      .provide([
+        [
+          select(navigationCurrentRouteSelector),
+          fromNullable(BONUSVACANZE_ROUTES.ACTIVATION.LOADING)
+        ],
+        [
+          matchers.call.fn(startBonusActivation),
```

Comment:
I found a bug by adding the case when the server returns something the app can't decode
```typescript
[
   matchers.call.fn(startBonusActivation),
   left([{ context: [], value: new Error("decoding failure") }])
]
```

I guess this scenario is also valuable.
The fix https://github.com/pagopa/io-app/pull/1931/commits/8568e5416fb0474971c56d54fa00dc93915b33e8

Reply:
I've changed the structure of `ActivationBackendResponse`, and now is possible to mock also the `left` scenario. I've added also the case you mentioned

``` typescript
const genericDecodingFailure = left([
  { context: [], value: new Error("decoding failure") }
]);
```